File: selfstudy/python/cities_counthigh.py
'''
task 177:
Cities - CountHigh
You need to count how many values are greater than 50 in city populations in thousands
'''

def count_above_fifty_cities(populations):
    return sum(1 for x in populations if x > 50)

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], count_above_fifty_cities([10, 20, 30])),
        ([50, 60, 70], count_above_fifty_cities([50, 60, 70])),
        ([1, 2, 3], count_above_fifty_cities([1, 2, 3])),
        ([100, 200], count_above_fifty_cities([100, 200])),
        ([], count_above_fifty_cities([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = count_above_fifty_cities(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# works i think

File: selfstudy/golang/cities_counthigh.go
/*
task 277:
Cities - CountHigh
You need to count how many values are greater than 50 in city populations in thousands
*/

package main

import (
	"fmt"
	"reflect"
)

func Count_above_fiftyCities(populations []int) int {
	count := 0
	for _, v := range populations {
		if v > 50 { count++ }
	}
	return count
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, Count_above_fiftyCities([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Count_above_fiftyCities([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Count_above_fiftyCities([]int{1, 2, 3})},
		{[]int{100, 200}, Count_above_fiftyCities([]int{100, 200})},
		{[]int{}, Count_above_fiftyCities([]int{})},
	}
	
	for idx, t := range tests {
		res := Count_above_fiftyCities(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done
